### PR TITLE
Documented the blas2 functions

### PR DESCRIPTION
**Pull request by RossBrunton**

### CODE REVIEW

File: include/interface/blas2_interface.h [hunk 1]
```diff
@@ -142,63 +250,174 @@ typename executor_t::policy_t::event_t inline _gemv(
...
+    executor_t& ex,     // executor_t (sycl, parallel, serial, etc)
+    char _Uplo,         // Whether the matrix is upper/lower ('u', 'l')
+    index_t _N,         // >0 The order of matrix A
+    element_t _alpha,   // Scalar alpha
+    container_0_t _vx,  // (1 + (_N-1)*abs(_incx)), input vector X
+    increment_t _incx,  // !=0 The increment for the elements of X
+    container_1_t _mA,  // (_lda, _N) The output matrix
+    index_t _lda        // >max(1, _N) The first dimension of _mA
+) {
   // TODO: Here we can use some heuristics to select localn global, local, and
```
Comment (mehdi-goli) [inline include/interface/blas2_interface.h]: remove this todo and instead create an internal issue

File: include/interface/blas2_interface.h [hunk 2]
```diff
@@ -142,63 +250,174 @@ typename executor_t::policy_t::event_t inline _gemv(
...
+    index_t _N,         // >0 The order of matrix A
+    element_t _alpha,   // Scalar alpha
+    container_0_t _vx,  // (1 + (_N-1)*abs(_incx)), input vector X
+    increment_t _incx,  // !=0 The increment for the elements of X
+    container_1_t _vy,  // (1 + (_N-1)*abs(_incx)), input vector Y
+    increment_t _incy,  // !=0 The increment for the elements of Y
+    container_2_t _mA,  // (_lda, _N) The output matrix
+    index_t _lda        // >max(1, _N) The first dimension of _mA
+) {
   // TODO: Here we can use some heuristics to select localn global, local, and
```
Comment (mehdi-goli) [inline include/interface/blas2_interface.h]: same as above

File: include/interface/blas2_interface.h (hunk 1 above)
Comment (RossBrunton) [reply include/interface/blas2_interface.h]: https://jira.office.codeplay.com/browse/SYCLE-717

File: include/interface/blas2_interface.h (hunk 2 above)
Comment (RossBrunton) [reply include/interface/blas2_interface.h]: https://jira.office.codeplay.com/browse/SYCLE-717

**Pull request merged**